### PR TITLE
Get time in enclave from host memory, not OCall

### PR DESCRIPTION
**Pull request by eddyashton**
Opening as a draft to see how this affects performance.

We have worker threads inside the enclave, and sometimes they want to idle because there is no work today. Currently they busy-wait with `__mm_pause`, which may do nothing, and wastes processor time. To actually idle, they need some concept of time, and a way to sleep. This PR adds that concept of time (with a uv job on the host writing time to a known memory location, which the enclave reads), and a sketch of backoff behaviour (busy-wait for a while, then start sleeping for chunks of time). If this works, I'll look at a more precise idle notification system (OCalling to wait on a `condition_variable` in host-space).

### CODE REVIEW

**Comment by achamayou** [comment]
This may be relevant: https://github.com/openenclave/openenclave/pull/2929

File: src/enclave/enclave.h [hunk 1]
```diff
@@ -245,7 +246,28 @@ namespace enclave
...
+          static std::chrono::microseconds idling_start_time;
+          const auto time_now = enclave::get_enclave_time();
+
+          if (consecutive_idles == 0)
+          {
+            idling_start_time = time_now;
+          }
+
+          // Handle initial idles by pausing, eventually sleep (in host)
+          constexpr std::chrono::milliseconds timeout(5);
```
Comment (olgavrou) [inline src/enclave/enclave.h:260]: do we want this to be configurable?

File: src/enclave/enclave_time.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  extern std::atomic<std::chrono::microseconds>* host_time;
+  extern std::chrono::microseconds last_value;
+
+  static std::chrono::microseconds get_enclave_time()
+  {
+    // Update cached value if possible, but never move backwards
+    if (host_time != nullptr)
+    {
+      const auto current_time = host_time->load();
+      if (current_time > last_value)
```
Comment (achamayou) [inline src/enclave/enclave_time.h:19]: We probably want to log this for now, make the count available to the operator later.

File: src/enclave/enclave.h (hunk 1 above)
Comment (eddyashton) [reply src/enclave/enclave.h:260]: In future yes, but we also want the sleep to do something smarter - I'll plumb in configurable parameters in a future PR.

**Pull request merged**